From 8191a26b9717f0b0b8beabdc765cc3c4713b2586 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Fri, 20 May 2011 18:06:00 -0400
Subject: [PATCH] Fix automagic libcap-ng dependency (GNOME bug #649936)

A slight modification of the patch by Saleem Abdulrasool
<compnerd@compnerd.org>. Like his patch, this is for 3.1.x.

Signed-off-by: Alexandre Rostovtsev <tetromino@gmail.com>
---
 configure.ac |   32 ++++++++++++++++++++++----------
 1 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index d9e947a..02372d8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -451,18 +451,30 @@ fi
 # libcap-ng
 #

-AC_CHECK_LIB([cap-ng], [capng_clear], have_libcapng="yes", have_libcapng="no")
-
-if test "$have_libcapng" = "yes"; then
-   AC_DEFINE(HAVE_LIBCAPNG, 1, [Have libcap-ng package, libcap-ng library])
-   DAEMON_LIBS="$DAEMON_LIBS -lcap-ng"
-else
-   have_lipcapng="no"
-   AC_MSG_WARN([libcap-ng (or development headers) is not installed])
+AC_ARG_WITH([libcap-ng],
+            [AC_HELP_STRING([--without-libcap-ng],
+                            [build without libcap-ng (disables Linux capabilities support)])],,
+            [with_libcap_ng=auto])
+
+if test x"$with_libcap_ng" != x"no" ; then
+    AC_CHECK_LIB([cap-ng], [capng_clear],
+                 [
+                   with_libcap_ng="yes"
+                   AC_DEFINE([HAVE_LIBCAPNG], [1], [have libcap-ng headers and library])
+                   DAEMON_LIBS="$DAEMON_LIBS -lcap-ng"
+                 ],
+                 [
+                   if test x"$with_libcap_ng" = x"yes" ; then
+                     AC_MSG_ERROR([libcap-ng support requested, but package not found])
+                   else
+                     AC_MSG_WARN([libcap-ng (or development headers) is not installed])
+                   fi
+                   with_libcap_ng="no"
+                 ])
 fi

-AM_CONDITIONAL(WITH_CAPS, test "$have_libcapng" = "yes")
-libcapng_status=$have_libcapng
+AM_CONDITIONAL([WITH_CAPS], [test x"$with_libcap_ng" = x"yes"])
+libcapng_status="$with_libcap_ng"

 # --------------------------------------------------------------------
 # Debug mode
--
1.7.5.rc3
